src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -66,38 +47,217 @@ internal static unsafe IPAddress Parse(string ipString, bool tryParse)
                 return null;
             }
 
-            Exception e = new SocketException(IPAddressPal.GetSocketErrorForErrorCode(error), error);
-            throw new FormatException(SR.dns_bad_ip_address, e);
+            throw new FormatException(SR.dns_bad_ip_address, new SocketException(SocketError.InvalidArgument));
         }
 
-        internal static string IPv4AddressToString(byte[] numbers)
+        internal static unsafe string IPv4AddressToString(uint address)
         {
-            StringBuilder sb = new StringBuilder(INET_ADDRSTRLEN);
-            uint errorCode = IPAddressPal.Ipv4AddressToString(numbers, sb);
+            const int MaxLength = 15;
+            char* addressString = stackalloc char[MaxLength];
+            int offset = MaxLength;
 
-            if (errorCode == IPAddressPal.SuccessErrorCode)
+            FormatIPv4AddressNumber((int)((address >> 24) & 0xFF), addressString, ref offset);
+            addressString[--offset] = '.';
+            FormatIPv4AddressNumber((int)((address >> 16) & 0xFF), addressString, ref offset);
```

Comment:
I dont understand this code, looks like the most significant address bit is put in least significant bit of the array, hence reversing the parts, but not reversing within the 8-bit parts? So, FF.00.00.01 becomes 01.00.00.FF, which means the given uint address bits are reversed?